test/tests/phi_zero/tests

```diff
@@ -0,0 +1,22 @@
+[Tests]
+  [./phi_zero_linear]
+    type = 'RunApp'
+    input = 'simple_transient_diffusion.i'
+    expect_out = ' Solve Converged!'
+    issues = '#15204'
+    design = ' '
+    requirement = 'The system shall be able to produce consestent size _phi_zero and _grad_phi_zero objects'
```

Comment:
We want our requirements to be as agnostic of implementation as possible. Maybe something more like:
```
requirement = 'The system shall be able to construct zero objects for shape functions and shape function gradients consistent with the maximum number of shape functions and quadrature points in the simulation.'
```